### PR TITLE
Otel Tag Option

### PR DESCRIPTION
**Pull request by Modularius**
## Summary of changes

Adds extra command line option, which prepends the "service.name" field for OpenTelemetry with an optional string. Useful for distinguishing telemetry from different instances of the pipeline running in parallel.

## Instruction for review/testing

General code review.

Tested on simulated data.

### CODE REVIEW

**Comment by DanNixon** [comment]
Is there a reason this modifies the service name rather than adding a new field/tag?

e.g. in the Prometheus world you would typically use a label for the name of the thing that is running and one or more to appropriately designate the specific instance of it.

**Comment by Modularius** [comment]
I initially thought of doing it that way, but I couldn't find a way of doing it that wouldn't be adding a tag to every single span. It might be that there is a way of doing this but this seemed like the most convenient thing to do in the meantime.

**Comment by Modularius** [comment]
Nothing. I think they are used for 3rd party tools like Jaeger rather than as user data.

**Comment by DanNixon** [comment]
`service.instance.id` seems to be what you want: https://opentelemetry.io/docs/specs/semconv/attributes-registry/service/

**Comment by Modularius** [comment]
Ah thanks, yes that would do. I think `service.namespace` is even better for this purpose.

File: nexus-writer/src/main.rs [hunk 1]
```diff
@@ -115,13 +119,14 @@ struct Cli {
 async fn main() -> anyhow::Result<()> {
     let args = Cli::parse();
 
+    debug!("{args:?}");
```
Comment (DanNixon) [inline nexus-writer/src/main.rs:122]: Should this be here?

File: nexus-writer/src/main.rs [hunk 2]
```diff
@@ -98,6 +98,10 @@ struct Cli {
     #[clap(long, default_value = "info")]
     otel_level: LevelFilter,
 
+    /// All OpenTelemetry Tags are appended with this string, if given. Can be used to track different instances of the pipeline running in parallel.
```
Comment (DanNixon) [inline nexus-writer/src/main.rs:101]: Do the doc comments need to be changed too?

**Review by DanNixon — CHANGES_REQUESTED** [review]

File: nexus-writer/src/main.rs (hunk 1 above)
Comment (Modularius) [reply nexus-writer/src/main.rs:122]: That's a relic from past logging, I'll remove it in a nexus-writer specific PR.

**Review by DanNixon — APPROVED** [review]

**Pull request merged**